### PR TITLE
VM Placement Changes for Terraform

### PR DESCRIPTION
**Pull request by ezilber-akamai**
## 📝 Description

Updated parameter names the VM Placement groups endpoints and updated documentation.
- "affinity_type" --> "placement_group_type"
- "is_strict" --> "placement_group_policy"

## ✔️ How to Test

Before running tests, set up the environment to test against alpha:
```
export LINODE_URL=api.dev.linode.com
export LINODE_API_VERSION=v4beta
export LINODE_TOKEN=...
```

Unit tests:
- `make unit-test PKG_NAME="linode/placementgroup"`

Integration tests:
- `make int-test PKG_NAME="linode/placementgroup"`
- `make int-test PKG_NAME="linode/placementgroups"`
- `make PKG_NAME=linode/instance ARGS="-run TestAccResourceInstance_withPG" int-test`
- `make PKG_NAME=linode/instance ARGS="-run TestAccResourceInstance_pgAssignment" int-test`

### CODE REVIEW

File: linode/instance/resource_test.go [hunk 1]
```diff
@@ -2382,13 +2382,14 @@ func TestAccResourceInstance_withPG(t *testing.T) {
...
-	if err != nil {
-		t.Fatal(err)
-	}
+	// // Resolve a region with support for PGs
+	// targetRegion, err := acceptance.GetRandomRegionWithCaps(
+	// 	[]string{"Linodes", "Placement Group"},
+	// )
+	// if err != nil {
+	// 	t.Fatal(err)
+	// }
```
[the remark this answers is not in the document]
Comment (ezilber-akamai) [reply linode/instance/resource_test.go:2391]: Meant to add it but forgot, good catch!

File: go.mod [hunk 2]
```diff
@@ -1,12 +1,14 @@
 module github.com/linode/terraform-provider-linode/v2
 
-go 1.21
+go 1.22.0
```
[the remark this answers is not in the document]
Comment (ezilber-akamai) [reply go.mod:3]: Good idea, I will update this.

[the remark this answers is not in the document]
Comment (ezilber-akamai) [reply go.mod:3]: When I try to run `go mod tidy -go=1.21` I get the error `go: k8s.io/api@v0.30.2 requires go@1.22.0, but 1.21 is requested` so it's possible other dependencies require 1.22.

File: linode/instance/resource_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ezilber-akamai) [reply linode/instance/resource_test.go:2392]: Fixed!

File: go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (ezilber-akamai) [reply go.mod]: Fixed!

File: go.sum [hunk 3]
```diff
@@ -185,8 +185,8 @@ github.com/kr/pty v1.1.1/go.mod h1:pFQYn66WHrOpPYNljwOMqo10TkYh1fy3cYio2l3bCsQ=
 github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
 github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
 github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
-github.com/linode/linodego v1.37.0 h1:B/2Spzv9jYXzKA+p+GD8fVCNJ7Wuw6P91ZDD9eCkkso=
-github.com/linode/linodego v1.37.0/go.mod h1:L7GXKFD3PoN2xSEtFc04wIXP5WK65O10jYQx0PQISWQ=
+github.com/lgarber-akamai/linodego v0.0.0-20240715152529-b3fb6f2dddff h1:WhDupkChfFWdF6PxeiplgAZ7fHJbKolThtzTW3IUnvM=
+github.com/lgarber-akamai/linodego v0.0.0-20240715152529-b3fb6f2dddff/go.mod h1:L7GXKFD3PoN2xSEtFc04wIXP5WK65O10jYQx0PQISWQ=
```
Comment (zliang-akamai) [inline go.sum:189]: Let's get linodego release and update it

Comment (ezilber-akamai) [reply go.sum]: Done!

**Review by lgarber-akamai — APPROVED** [review]
Looks perfect, great work!

**Pull request merged**